crates/oxc_allocator/src/vec.rs

```diff
@@ -167,44 +165,10 @@ impl<'alloc, T> Vec<'alloc, T> {
         let vec = unsafe { InnerVec::from_raw_parts_in(ptr, N, N, allocator.bump()) };
         Self(ManuallyDrop::new(vec))
     }
-
-    /// Converts the vector into [`Box<[T]>`][owned slice].
-    ///
-    /// Any excess capacity the vector has will not be included in the slice.
-    /// The excess memory will be leaked in the arena (i.e. not reused by another allocation).
-    ///
-    /// # Examples
-    /// ```
-    /// use oxc_allocator::{Allocator, Vec};
-    ///
-    /// let allocator = Allocator::default();
-    /// let mut v = Vec::with_capacity_in(10, &allocator);
-    /// v.extend([1, 2, 3]);
-    /// let b = v.into_boxed_slice();
-    ///
-    /// assert_eq!(&*b, &[1, 2, 3]);
-    /// assert_eq!(b.len(), 3);
-    /// ```
-    ///
-    /// [owned slice]: Box
-    #[inline]
-    pub fn into_boxed_slice(self) -> Box<'alloc, [T]> {
-        let inner = ManuallyDrop::into_inner(self.0);
-        let slice = inner.leak();
-        let ptr = NonNull::from(slice);
-        // SAFETY: `ptr` points to a valid slice `[T]`.
-        // `allocator_api2::vec::Vec::leak` consumes the inner `Vec` without dropping it.
-        // Lifetime of returned `Box<'alloc, [T]>` is same as lifetime of consumed `Vec<'alloc, T>`,
-        // so data in the `Box` must be valid for its lifetime.
-        // `Vec` uniquely owned the data, and we have consumed the `Vec`, so the new `Box` has
-        // unique ownership of the data (no aliasing).
-        // `ptr` was created from a `&mut [T]`.
-        unsafe { Box::from_non_null(ptr) }
-    }
```

Comment:
This method does not work directly after replacing it with `Vec2`. Since there no real use cases, so let's remove it instead first

Reply:
Can we do that in a separate PR? As this method is not used, it wouldn't even need to be part of this stack.

Reply:
> Can we do that in a separate PR? As this method is not used, it wouldn't even need to be part of this stack.

#9735